lib/units/device/plugins/filesystem.js

```diff
@@ -0,0 +1,86 @@
+var util = require('util')
+
+var syrup = require('stf-syrup')
+var adbkit = require('adbkit')
+var path = require('path')
+
+var logger = require('../../../util/logger')
+var wire = require('../../../wire')
+var wireutil = require('../../../wire/util')
+
+module.exports = syrup.serial()
+  .dependency(require('../support/adb'))
+  .dependency(require('../support/router'))
+  .dependency(require('../support/push'))
+  .dependency(require('../support/storage'))
+  .dependency(require('../resources/minicap'))
+  .dependency(require('./util/display'))
+  .define(function(options, adb, router, push, storage, minicap, display) {
+    var log = logger.createLogger('device:plugins:filesystem')
+    var plugin = Object.create(null)
+
+    plugin.retrive = function(file) {
+      log.info('Retriving file %s', file)
+
+      return adb.pull(options.serial, file)

```

Comment:
Looks like you're pulling the same file twice (once more later), is there a reason for that?